api/test/wfl/integration/source_test.clj

```diff
@@ -37,18 +47,86 @@
 
 (defn ^:private create-tdr-source []
   (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-    (->> {:name           "Terra DataRepo",
-          :dataset        "this"
-          :table          "is"
-          :column         "fun"
-          :skipValidation true}
+    (->> {:column         "a-column"
+          :dataset        {:name "a-dataset"}
+          :id             "an-id"
+          :name           "Terra DataRepo"
+          :skipValidation true
+          :table          "a-table"
+          :type           "a-type"}
          (source/create-source! tx (rand-int 1000000))
          (zipmap [:source_type :source_items])
          (source/load-source! tx))))
 
 (defn ^:private reload-source [tx {:keys [type id] :as _source}]
   (source/load-source! tx {:source_type type :source_items (str id)}))
 
+;; A stable vector of TDR row IDs.
+;;
+(defonce all-rows
+  (delay (vec (repeatedly mock-new-rows-size #(str (UUID/randomUUID))))))
+
+(defn ^:private mock-query-table-between-all
+  "Mock datarepo/query-table-between to find all rows."
+  [_dataset _table _between interval columns]
+  (is (every? parse-timestamp interval))
+  (letfn [(field [column] {:mode "NULLABLE" :name column :type "STRING"})
+          (fields [columns] (mapv field columns))]
+    (-> {:jobComplete true
+         :kind "bigquery#queryResponse"
+         :rows [@all-rows]
+         :totalRows (str (count @all-rows))}
+        (assoc-in [:schema :fields] (fields columns)))))
+
+(defn ^:private mock-query-table-between-miss
+  "Mock datarepo/query-table-between to miss some rows."
+  [dataset table between interval columns]
+  (-> (mock-query-table-between-all dataset table between interval columns)
+      (update-in [:rows 0] butlast)
+      (assoc :totalRows (str (dec mock-new-rows-size)))))
+
+;;234567890123456789012345678901234567890123456789012345678901234567890123456789
```

Comment:
What is this?

Reply:
Looks like two semi-colons followed by a string of digits. ;-0

Reply:
I'm glad your having fun

Reply:
Sad that you are not?